deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/Deployer.java

```diff
@@ -91,8 +92,29 @@ public default void waitForDeployment(DeploymentContext context, ProgressHandler
 	 */
 	public List<DeploymentStatus> getDeploymentStatus(DeploymentContext deploymentContext) throws HyscaleException;
 
-	
-	public List<ReplicaInfo> getReplicas(DeploymentContext deploymentContext, boolean isFilter) throws HyscaleException;
+
+	/**
+	 * Get replicas info based on pods
+	 * <p>
+	 * isFilter if disabled return replica info for all pods fetched based on selector
+	 * if enabled,
+	 * 1. If owner kind for pods is different, warn user and return replica info for all pods
+	 * 2. If owner kind is {@link ResourceKind #DEPLOYMENT} or {@link ResourceKind #REPLICA_SET},
+	 *     Get Revision from deployment, get replicas set with this revision,
+	 *     filter pods based on pod-template-hash from replica set
+	 *     return replica info for filtered pods
+	 * 3. Else return replica info for all pods
+	 * </p>
```

Comment:
This explanation should be in the implementation, as it is an interface and abstracts out all the Kubernetes details here.